opensrp-reporting/src/main/java/org/smartregister/reporting/impl/ReportingModule.java

```diff
@@ -0,0 +1,16 @@
+package org.smartregister.reporting.impl;
+
+import android.view.ViewGroup;
+
+import org.smartregister.reporting.domain.IndicatorTally;
+
+import java.util.List;
+import java.util.Map;
+
+public interface ReportingModule {
```

Comment:
Not sure about this abstraction because the `generateReport()` looks like it could exist as a Util function instead if required

Reply:
I do not think a utility function is the right approach in this case. The idea was to separate the way we create the visualizations. Splitting their implementation is necessary when you are dealing with more than just one indicator defined in the one **yaml** configuration. Having different implementations implement this contract allows for consistency. Otherwise the views will just be clustered in the activity/fragment that they are created in and becomes had to  maintain.